python_modules/dagster/dagster/_core/definitions/asset_checks.py

```diff
@@ -29,15 +31,29 @@
 if TYPE_CHECKING:
     from dagster._core.definitions.assets import AssetsDefinition
 
+TPropsType = TypeVar("TPropsType", bound=NamedTuple)
+TDefType = TypeVar("TDefType")
 
-@experimental
-class AssetChecksDefinitionInputOutputProps(NamedTuple):
+
+class IDefinitionPropsClassMatchesInit(Generic[TPropsType, TDefType], ABC):
```

Comment:
nit: something in the direction of `IDefinitionWithProps` / `PropsBackedDefinition`, i think it makes more sense to target the fundamentlal pattern then a specific thing we enforce when participating in it